core/src/main/java/io/fabric8/maven/core/service/ApplyService.java

```diff
@@ -413,7 +418,80 @@ public void applyPersistentVolumeClaim(PersistentVolumeClaim entity, String sour
+                    try {
+                        HasMetadata answer = patchService.compareAndPatchEntity(namespace, entity, old);
+                        log.info("Updated Custom Resource Definition result: " + getName(answer));
+                    } catch (Exception e) {
+                        onApplyError("Failed to update Custom Resource Definition from " + sourceName + ". " + e + ". " + entity, e);
+                    }
+                }
+            }
+        } else {
+            if (!isAllowCreate()) {
+                log.warn("Creation disabled so not creating a Custom Resource Definition from " + sourceName + " name " + getName(entity));
+            } else {
+                doCreateCustomResourceDefinition(entity, sourceName);
+            }
+        }
+    }
+
+    private void doCreateCustomResourceDefinition(CustomResourceDefinition entity, String sourceName) {
+        log.info("Creating a Custom Resource Definition from " + sourceName + " name " + getName(entity));
+        try {
+            Object answer = kubernetesClient.customResourceDefinitions().create(entity);
+            log.info("Created Custom Resource Definition result: " + ((CustomResourceDefinition) answer).getMetadata().getName());
+        } catch (Exception e) {
+            onApplyError("Failed to create Custom Resource Definition from " + sourceName + ". " + e + ". " + entity, e);
+        }
+    }
+
+    public void applyCustomResource(File customResourceFile, String namespace, CustomResourceDefinitionContext context) throws Exception {
+        Map<String, Object> cr = KubernetesClientUtil.doReadCustomResourceFile(customResourceFile);
+        String name = (String)((Map<String, Object>)cr.get("metadata")).get("name");
+
+        if (isRecreateMode()) {
+            KubernetesClientUtil.doDeleteCustomResource(kubernetesClient, context, namespace, name);
+            KubernetesClientUtil.doCreateCustomResource(kubernetesClient, context, namespace, customResourceFile);
+            log.info("Created Custom Resource: " + name);
+        } else {
+            cr = KubernetesClientUtil.doGetCustomResource(kubernetesClient, context, namespace, name);
+            if (cr == null) {
+                KubernetesClientUtil.doCreateCustomResource(kubernetesClient, context, namespace, customResourceFile);
+                log.info("Created Custom Resource: " + name);
+            } else {
+                KubernetesClientUtil.doEditCustomResource(kubernetesClient, context, namespace, name, customResourceFile);
+                log.info("Updated Custom Resource: " + name);
+            }
+        }
+    }
+
+    public void deleteCustomResource(File customResourceFile, String namespace, CustomResourceDefinitionContext crdContext) throws Exception {
+        Map<String, Object> customResource = KubernetesClientUtil.doReadCustomResourceFile(customResourceFile);
+        String name = (String)((Map<String, Object>)customResource.get("metadata")).get("name");
```

Comment:
Can you try to put this statement in two lines, too much parenthesis and casts in just one line :)